index.js

```diff
@@ -40,29 +43,43 @@ client.on('ready', () => {
 // interaction has type: ChatInputCommandInteraction
 // https://old.discordjs.dev/#/docs/discord.js/main/class/ChatInputCommandInteraction
 client.on('interactionCreate', async interaction => {
-  if (!interaction.isChatInputCommand()) return;
-
   // defines the action to take when user enters /ping
-  console.log(interaction);
+  if (!interaction.isChatInputCommand()) return;
 
-  if (interaction.commandName === 'ping') {
-    await ping.execute(interaction);
-  }
+  const command = interaction.client.commands.get(interaction.commandName);
 
-  if (interaction.commandName === 'echo') {
-    await echo.execute(interaction);
+  if (!command) {
+    console.error(`No command matching ${interaction.commandName} was found.`);
+    return;
   }
 
-  if (interaction.commandName === 'queue') {
-    await queue.execute(interaction);
+  try {
+    await command.execute(interaction);
+  } catch (error) {
+    console.error(error);
+    await (interaction.replied || interaction.deferred
+      ? interaction.followUp({
+          content: 'There was an error while executing this command!',
+          ephemeral: true,
+        })
+      : interaction.reply({
+          content: 'There was an error while executing this command!',
+          ephemeral: true,
+        }));
   }
 });
 
 async function main() {
```

Comment:
We should have debug states to encourage testing and non debug states, which allows us to remove all the debug code at once later,